### DISCUSSION TITLE
Editable customer ID

### DISCUSSION THREAD

**Discussion by maurice-kpw** [kimai/kimai#3369] [Questions & Answers] [answered]
Hello!

I want to show the customer ID on the created invoice. Currently I'm using ${customer.id} in a word template. This is displaying the automatically assigned ID from kimai. Since we already have customers with existing unique IDs it would be great to either have the field "Customer ID" to fill out or to be able to edit this existing ID. 

Or do I need to use custom meta fields to customers for this?

Thank you!
– Maurice

**Comment by kevinpapst** [comment, marked as answer]
Use the account field

**Reply by maurice-kpw** [reply]
Would it be possible to have the account number generated automatically similar to invoice-numbers? So the field is automatically populated with a number generated by "{M}{y}-{cy,4}" for example?

**Reply by kevinpapst** [reply]
You can open a feature request for that. But please include exact details what you think are useful replacer and how they should work.
For myself: I can only think of a flag for the length of the auto-generated number.

**Reply by maurice-kpw** [reply]
Will do!

Edit: Opened feature request here: https://github.com/kevinpapst/kimai2/discussions/3372

**Comment by maurice-kpw** [comment]
Wow that was quick! Thank you :) 

That worked great. I was a little confused, because "Account" doesn't show up in the customer variables (https://www.kimai.org/documentation/invoices.html#customer-variables) but i figured it was ${customer.number}. 

Thanks!

**Reply by kevinpapst** [reply]
Weird, you seem to have a different docs than I do 😁 
<img width="742" alt="Bildschirmfoto 2022-06-22 um 16 02 21" src="https://user-images.githubusercontent.com/533162/175048614-ff5f0652-dcda-4663-93e2-487cc5e65ec2.png">

**Reply by kevinpapst** [reply]
Ah right, now I understand. Thanks, will update the docs to reflect the term "account"

**Reply by kevinpapst** [reply]
Can you do me a favor and check https://www.kimai.org/documentation/invoices.html?sdf#customer-variables and let me know if there is more that you would like to see improved

**Reply by maurice-kpw** [reply]
Yes! I see the changes. Great idea to highlight the ui-labels :) If I have further suggestions I'll let you know!
